### PR TITLE
Start of extended babel transform for import.meta.glob(Eager)

### PR DESCRIPTION
**Pull request by rubenmoya**
This PR adds a babel transform to make `import.meta.glob` work.

### CODE REVIEW

File: jest/babel-test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+`;
+
+const result = babel.transform(code, {
+  filename: "./src/file.ts",
+  plugins: [importMetaBabelPlugin],
+});
+
+console.log("\n--------------------------\n");
+
+console.log(result.code);
```
Comment (mpeyper) [inline jest/babel-test.js]: We will need to delete this file eventually, but it's handy while we develop the plugin.

File: jest/importMetaBabelPlugin.js [hunk 2]
```diff
@@ -20,8 +22,34 @@ module.exports = function importMetaBabelPlugin() {
     },
   })
 `);
+
+  // const modules = {
+  //   './dir/foo.js': () => import('./dir/foo.js'),
+  //   './dir/bar.js': () => import('./dir/bar.js')
+  // }
```
Comment (mpeyper) [inline jest/importMetaBabelPlugin.js]: We can remove these comments now, was just to remind me what the output was supposed to look like without constantly switching back to the vite docs.

File: jest/babel-test.js (hunk 1 above)
Comment (mpeyper) [reply jest/babel-test.js]: Or maybe we should convert it into a jest test and actually assert to transformed code 🤔

Comment (rubenmoya) [reply jest/babel-test.js]: I removed the file, we could bring it back and add a test or wait until we create a package with this babel transform

**Review by nobrayner — APPROVED** [review]
LGTM (for all the good that will do us)

**Pull request merged**